lib/src/erc20/systems/approval/erc20_approve.cairo

```diff
@@ -0,0 +1,11 @@
+#[system]
+mod ERC20_Approve {
+    execute(token_id: felt252, spender: ContractAddress, amount: u256) {
+        let caller = get_caller_address();
+        let approval_sk: StorageKey = (token_id, (caller, spender)).into();
```

Comment:
```suggestion
        let approval_sk: StorageKey = (token_id, (caller.into(), spender)).into();
```

you'll need to convert `ContractAddress` into a `felt252`. You might have to import the trait too